### PR TITLE
[repo] Bump Microsoft.Extensions.Logging.Configuration and clean up net462 targets

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #5015

## Changes

* Updates Microsoft.Extensions.Logging.Configuration to .NET 8 RC2.
* Cleans up "net462" being sprinkled around all over the repo.
* Uses `'$(TargetFrameworkIdentifier)' == '.NETCoreApp'` style in more places.

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes

### CODE REVIEW

File: src/Shared/HttpSemanticConventionHelper.cs [hunk 1]
```diff
@@ -77,15 +77,15 @@ private static bool TryGetConfiguredValues(IConfiguration configuration, [NotNul
     {
         try
         {
-            var stringValue = configuration![SemanticConventionOptInKeyName];
+            var stringValue = configuration[SemanticConventionOptInKeyName];
```
Comment (utpilla) [inline src/Shared/HttpSemanticConventionHelper.cs:80]: Why remove the `!` here?

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj [hunk 2]
```diff
@@ -28,4 +28,8 @@
     <PackageReference Include="OpenTelemetry" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="System.Net.Http" Condition="'$(TargetFramework)' == '$(NetFrameworkMinimumSupportedVersion)'" />
+  </ItemGroup>
```
Comment (TimothyMothra) [inline src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj:33]: Why this change?
Looks like this wasn't included before. 
https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http/1.6.0-beta.2#dependencies-body-tab

File: src/Shared/HttpSemanticConventionHelper.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/Shared/HttpSemanticConventionHelper.cs:80]: Just didn't seem to be needed. Doesn't warn without it so not sure why it was there in the first place 🤷

File: src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj (hunk 2 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj:33]: Good eye there @TimothyMothra! Not totally sure. This baffled me and @utpilla a bit. We used to reference the 3.1 versions of these packages. Interestingly they have a `netstandard2.0` target but NOT a `net462` target. We are jumping to the .NET 8 RC versions which do have a `net462` target. For some reason, using that new dedicated `net462` target a bunch of these projects started complaining about missing `System.Net.Http`! My theory is that `HttpClient` is part of `netstandard2.0` but NOT `net462`. It seems like when the `netstandard2.0` bits were used the whole .NET Standard 2.0 API layer came with it including `HttpClient`. Best guess 😆 If you want to spend some time on it and try to figure it out I am very curious to know exactly what is going on but I decided not enough so to keep spending time on it.

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.Http/OpenTelemetry.Instrumentation.Http.csproj:33]: > It seems like when the netstandard2.0 bits were used the whole .NET Standard 2.0 API layer came with it including HttpClient.

Yes, that's indeed the case here. This screenshot is from dotPeek, which shows that the `netstandard2.0` bits for this package bring in `netstandard2.0` APIs:

image
